modules/aws/lambda.go

```diff
@@ -2,51 +2,136 @@ package aws
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 
 	"github.com/aws/aws-sdk-go/service/lambda"
 	"github.com/gruntwork-io/terratest/modules/testing"
 	"github.com/stretchr/testify/require"
 )
 
+// LambdaOptions contains additional parameters for InvokeFunctionWithParams().
+// It contains a subset of the fields found in the lambda.InvokeInput struct.
+type LambdaOptions struct {
+	// FunctionName is a required field containing the lambda function name.
+	FunctionName *string
+
+	// InvocationType can be one of "RequestResponse" or "DryRun".
+	//    * RequestResponse (default) - Invoke the function synchronously.
+	//    Keep the connection open until the function returns a response
+	//    or times out.
+	//
+	//    * DryRun - Validate parameter values and verify that the user or
+	//    role has permission to invoke the function.
+	InvocationType *string
+
+	// Lambda function input; will be converted to JSON.
+	Payload interface{}
+}
+
+// LambdaOutput contains the output from InvokeFunctionWithParams().  The
+// fields may or may not have a value depending on the invocation type and
+// whether an error occurred or not.
+type LambdaOutput struct {
+	// If present, indicates that an error occurred during function execution.
+	// Error details are included in the response payload.
+	FunctionError *string
+
+	// The response from the function, or an error object.
+	Payload []byte
+
+	// The HTTP status code for a successful request is in the 200 range.
+	// For RequestResponse invocation type, the status code is 200.
+	// For the DryRun invocation type, the status code is 204.
+	StatusCode *int64
+}
+
 // InvokeFunction invokes a lambda function.
 func InvokeFunction(t testing.TestingT, region, functionName string, payload interface{}) []byte {
-	out, err := InvokeFunctionE(t, region, functionName, payload)
+	input := &LambdaOptions{
+		FunctionName: &functionName,
+		Payload:      &payload,
+	}
+	out, err := InvokeFunctionWithParams(t, region, input)
 	require.NoError(t, err)
-	return out
+	return out.Payload
```

Comment:
This should call `InvokeFunctionE`. No need to change it.

Reply:
Done.